pkg/metrics/metrics.jsx

```diff
@@ -390,20 +404,33 @@ class CurrentMetrics extends React.Component {
                 </Tooltip>);
         }
 
+        const top_cores = this.state.cpuCoresUsed.map((v, i) => [i, v]).sort((a, b) => b[1] - a[1])
+                .slice(0, 16);
+        const cores = top_cores.map(c => <Flex key={c[0]} justifyContent={{ default: 'justifyContentSpaceBetween' }}>
+            <FlexItem>{ cockpit.format(_("Core $0"), c[0]) }</FlexItem>
+            <FlexItem>{c[1]}%</FlexItem></Flex>
+        );
+
+        const cpu_usage = (
+            <Progress
+                id="current-cpu-usage"
+                value={this.state.cpuUsed}
+                className="pf-m-sm"
+                min={0} max={100}
+                variant={ this.state.cpuUsed > 90 ? ProgressVariant.danger : ProgressVariant.info }
+                title={ num_cpu_str }
+                label={ this.state.cpuUsed + '% ' } />);
+
         return (
             <Gallery className="current-metrics" hasGutter>
                 <Card id="current-metrics-card-cpu">
                     <CardTitle>{ _("CPU") }</CardTitle>
                     <CardBody>
                         <div className="progress-stack">
-                            <Progress
-                                id="current-cpu-usage"
-                                value={this.state.cpuUsed}
-                                className="pf-m-sm"
-                                min={0} max={100}
-                                variant={ this.state.cpuUsed > 90 ? ProgressVariant.danger : ProgressVariant.info }
-                                title={ num_cpu_str }
-                                label={ this.state.cpuUsed + '% ' } />
+                            {this.state.cpuCoresUsed.length
```

Comment:
I'd change this to `> 1`, it doesn't make much sense to do that on 1-core machines (which are still very common for VMs). Then you  could also conditionalize the computation of `cores` and skip it on single-core machines.